public/modules/custom/grants_attachments/src/AttachmentFixerService.php

```diff
@@ -0,0 +1,251 @@
+      }
+    }
+
+    // Make sure we have labels for all fields.
+    foreach ($attachmentInfo as $key => $info) {
+      foreach ($info as $key2 => $item) {
+        if (!isset($item['label'])) {
+          // If not, use the ID as label.
+          $attachmentInfo[$key][$key2]['label'] = $attachmentInfo[$key][$key2]['ID'];
+        }
+      }
+    }
+
+    $content['attachmentsInfo']['attachmentsArray'] = $attachmentInfo;
+    $atvDoc->setContent($content);
+    return $atvDoc;
+  }
+
+  /**
+   * If integrationID is missing from application data, add it.
+   *
+   * @param array $attachment
+   *   The attachment.
+   * @param array $attachmentInfo
+   *   The attachment info.
+   * @param string $appEnv
+   *   The application environment for this application.
+   *
+   * @return void
+   *   Void.
+   */
+  private function updateIntegrationIdForAttachment(array $attachment, array &$attachmentInfo, string $appEnv): void {
+    // Clean file href to integrationID format so that it'll work with ATV.
+    $intID = '/' . $appEnv . AttachmentHandlerHelper::cleanIntegrationId($attachment['href']);
+
+    // Loop attachment fields that are added to the application.
+    foreach ($attachmentInfo as &$innerArray) {
+      $fileNameMatched = FALSE;
+      $integrationIdUpdated = FALSE;
+
+      foreach ($innerArray as &$item) {
+        if ($item['ID'] === 'fileName' && $item['value'] === $attachment['filename']) {
+          $fileNameMatched = TRUE;
+        }
+        if ($fileNameMatched && $item['ID'] === 'integrationID') {
+          // Update integrationID in place.
+          $item['value'] = $intID;
+          // Set the value to control adding new integrationID.
+          $integrationIdUpdated = TRUE;
+          break;
+        }
+      }
+      // If filename matched but no integrationID was found, add it.
+      if ($fileNameMatched && !$integrationIdUpdated) {
+        $innerArray[] = [
+          'ID' => 'integrationID',
+          'label' => 'Integration ID',
+          'value' => $intID,
+          'valueType' => 'string',
+          'meta' => "[]",
+        ];
+      }
+    }
+  }
+
+  /**
+   * Try to figure our if attachments are ok.
+   *
+   * This is a helper function for fixAttachmentsOnApplication that goes through
+   * the events and attachment info to see if the attachments are added properly
+   * to the document.
+   *
+   * @param mixed $events
+   *   Events.
+   * @param mixed $attachment
+   *   Attachment.
+   * @param mixed $attachmentInfo
+   *   Attachment info.
+   * @param mixed $appEnv
+   *   Application environment.
+   *
+   * @return array
+   *   Array of booleans.
+   */
+  public function areAttachmentsOk(
+    mixed $events,
+    mixed $attachment,
+    mixed $attachmentInfo,
+    mixed $appEnv,
+  ): array {
+    // Look for events from us, the handler.
+    $handlerOk = $this->filterEventsByTypeAndFilename($events, 'HANDLER_ATT_OK', $attachment['filename']);
```

Comment:
`filterEventsByTypeAndFilename` expects `$events` and `$attachment` to be an array, but this method allows `mixed` type for both. The type should be checked/enforced before calling the method.